### PR TITLE
Exposes more of AccountData to AVM

### PR DESCRIPTION
**Pull request by jannotti**
Getting this on our radar. I think exposing these fields would help apps that want to keep an eye on their MBR.

The only thing that is potentially worth exposing now is `Status` I'd rather ignore it for now (see explanation in a comment below). Debate me!

### CODE REVIEW

**Review by jasonpaulos — COMMENTED** [review]
Looks pretty good, I just have some comments.

File: data/transactions/logic/box_test.go [hunk 1]
```diff
@@ -331,6 +331,27 @@ func TestConveniences(t *testing.T) {
...
+
+	ledger.NewApp(txn.Sender, 888, basics.AppParams{})
+	// The SENDER certainly has no boxes (but does exist)
+	logic.TestApp(t, `int 0; acct_params_get AcctTotalBoxes; pop; !`, ep)
+	// Nor does the app account, to start
+	logic.TestApp(t, `int 888; app_params_get AppAddress; assert;
+	                  acct_params_get AcctTotalBoxes; pop; !; `, ep)
+	// Create a 31 byte box with a 4 byte name
+	logic.TestApp(t, `byte "self"; int 31; box_create`, ep)
+	fmt.Printf("%+v\n", ep.Ledger)
```
[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/box_test.go:348]: nope

File: data/transactions/logic/fields.go [hunk 2]
```diff
@@ -1205,6 +1205,24 @@ const (
...
+	// AcctTotalAssetsCreated is the number of ASAs created by this account
+	AcctTotalAssetsCreated
+	// AcctTotalAssets is the number of ASAs opted in by this account (always includes AcctTotalAssetsCreated)
+	AcctTotalAssets
+	// AcctTotalBoxes is the number of boxes created by the app this account is associated with
+	AcctTotalBoxes
+	// AcctTotalBoxBytes is the number of bytes in all boxes of this app account
+	AcctTotalBoxBytes
+
+	// AcctTotalAppSchema - consider how to expose
```
[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/fields.go:1224]: I agree, that'd work.  I'm willing if you (or anyone else) is willing to say "Do it."

Else it'll wait until someone cares about it.

[the remark this answers is not in the document]
Comment (jasonpaulos) [reply data/transactions/logic/fields.go:1224]: "Do it."

[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/fields.go:1229]: "Did it"

[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/fields.go:1229]: So now should I expose `Status`? It's the only thing left I would consider exposing (the other stuff is voting keys and rewards info that is annoying to explain).

**Comment by jannotti** [comment]
I'm going to merge this to the feature branch without doing anything about `Status` around 4pm unless I hear objections.  Objections should express an opinion about how it should be exposed.

File: data/transactions/logic/fields.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasonpaulos) [reply data/transactions/logic/fields.go:1229]: I agree with your reasons not to expose `Status` at this time.

**Review by jasonpaulos — APPROVED** [review]
This looks good!

**Pull request merged**